### ISSUE TITLE
Actor EditorActorComponent.cpp could use a refactor organizing with Controller and Config classes

### ISSUE THREAD

**Issue by smurly** [o3de/o3de#9918]
**Is your feature request related to a problem? Please describe.**
comparing the EditorActorComponent.cpp implementation of UI and editor python bindings reflection it could use some reorganization and a shift to a standard more like EditorMeshComponent.cpp
editContext->Class<EditorActorComponent> seems to contain all attributes and dataelements stuffed together
the exposed Editor Python Bindings property paths are root level on the component and their organization is hard to follow.

property names should follow a consistent capitalization. either Mesh should have "Model asset" or Actor should have "Actor Asset" (Asset is capitalized currently in Mesh but lower case in Actor)

a non-UI visible property path includes "Automatically update bounds?" which should not have a question mark in it and seems like an odd property to expose to reflected bindings. Don't put punctuation or question marks in DataElement name strings.

for reference these are the Actor property paths with type I pulled from the component via python bindings property tree editor
m_materialPerActor serializes as SimpleAssetReference<AssetType><MaterialAsset > which is not a python compatible type it seems like this property should not be reflected to python bindings at all, like it's coming from an associated material component.
```
'Out of view|Bounding box configuration|Automatically update bounds?' (bool,NotVisible)
'Actor asset' ('Asset', 'Visible')
'Out of view|Bounding box configuration' ('BoundingBoxConfiguration', 'Visible')
'Out of view|Bounding box configuration|Bounds type' ('unsigned char', 'Visible')
'Attach To|Attachment type' ('unsigned int', 'Visible')
'Attach To|Target entity' ('EntityId', 'NotVisible')
'Render options|Draw character' ('bool', 'Visible')
'Render options|Draw bounds' ('bool', 'Visible')
'Out of view|Bounding box configuration|Update frequency' ('float', 'NotVisible')
'Out of view|Bounding box configuration|Expand by' ('float', 'Visible')
'Material' ('SimpleAssetReference<AssetType><MaterialAsset >', 'NotVisible')
'Render options|Draw skeleton' ('bool', 'Visible')
'Render options|Skinning method' ('unsigned int', 'Visible')
'Out of view|Force update joints' ('bool', 'Visible')
'Out of view|Bounding box configuration|Update item skip factor' ('unsigned int', 'NotVisible')
```

**Describe the solution you'd like**
organization should use additional class definitions for organizing UI and controls. line indentation should be used to establish scope ownership of attributes. the following code snippet is more just a loose suggestion of organization since I focus more on the python side
```
editContext->Class<EditorActorComponent>("Actor", "The Actor component manages an instance of an Actor")
                        ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
                            ->Attribute(AZ::Edit::Attributes::Category, "Animation")
                            ->Attribute(AZ::Edit::Attributes::Icon, ":/EMotionFX/ActorComponent.svg")
                            ->Attribute(AZ::Edit::Attributes::PrimaryAssetType, azrtti_typeid<ActorAsset>())
                            ->Attribute(AZ::Edit::Attributes::ViewportIcon, ":/EMotionFX/Viewport/ActorComponent.svg")
                            ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game", 0x232b318c))
                            ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
                            ->Attribute(AZ::Edit::Attributes::HelpPageURL, "https://o3de.org/docs/user-guide/components/reference/animation/actor/")
                        ->UIElement(AZ::Edit::UIHandlers::Button, "Add Material Component", "Add Material Component")
                            ->Attribute(AZ::Edit::Attributes::NameLabelOverride, "")
                            ->Attribute(AZ::Edit::Attributes::ButtonText, "Add Material Component")
                            ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorActorComponent::AddEditorMaterialComponent)
                            ->Attribute(AZ::Edit::Attributes::Visibility, &EditorActorComponent::GetEditorMaterialComponentVisibility)

editContext->Class<ActorComponentController>()
editContext->Class<ActorComponentConfig>(
                        "ActorComponentConfig", "")
                        ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
                            ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
                            ->DataElement(0, &EditorActorComponent::m_actorAsset, "Actor asset", "Assigned actor asset")
                                ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorActorComponent::OnAssetSelected)
                                ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
                                ->Attribute(AZ_CRC_CE("EditButton"), "")
                                ->Attribute(AZ_CRC_CE("EditDescription"), "Open in Animation Editor")
                                ->Attribute(AZ_CRC_CE("DisableEditButtonWhenNoAssetSelected"), true)
                                ->Attribute("EditCallback", &EditorActorComponent::LaunchAnimationEditor)
```

**Describe alternatives you've considered**

**Additional context**
Ultimately you do you. this component just seemed very unusually organized having worked mostly with python bindings for atom components.

**Comment by lgleim** [comment]
@smurly Thank you for raising this request. We warmly welcome contributions on this matter.
